### PR TITLE
Support RSA double certs and RSA_SM4 cipher suites for NTLS

### PR DESCRIPTION
**Pull request by dongbeiouba**
Both SM2 and RSA double certs are supported for NTLS.

Add RSA_SM4_CBC_SM3, RSA_SM4_GCM_SM3, RSA_SM4_CBC_SHA256 and RSA_SM4_GCM_SHA256 cipher suites for NTLS.

Fix incorrect use of OPENSSL_NO_SM2/3/4 macros.

Add test/certs/dc/*.dc, *.key, *.crt and test/certs/sm2/*.crt, *.key to eliminate dependencies between test cases.

### CODE REVIEW

**Review by InfoHunter — APPROVED** [review]

File: ssl/s3_lib.c [hunk 1]
```diff
@@ -2950,12 +2950,14 @@ static SSL_CIPHER ssl3_ciphers[] = {
...
-     && (!defined OPENSSL_NO_SM3) && (!defined OPENSSL_NO_SM4)
+#ifndef OPENSSL_NO_NTLS
+# ifndef OPENSSL_NO_SM4
+#  ifndef OPENSSL_NO_SM3
+#   ifndef OPENSSL_NO_SM2
     {
      1,
      NTLS_TXT_ECDHE_SM2_SM4_CBC_SM3,
-     NULL,
+     NTLS_GB_ECDHE_SM2_SM4_CBC_SM3,
```
Comment (InfoHunter) [inline ssl/s3_lib.c:2960]: 为啥叫GB？

File: ssl/ssl_rsa.c [hunk 2]
```diff
@@ -160,12 +154,29 @@ static int ssl_use_certificate_file_idx(SSL *ssl, const char *file,
...
+    }
+
+    pkey = X509_get0_pubkey(x);
+    if (pkey == NULL) {
+        SSLerr(SSL_F_SSL_USE_CERTIFICATE_FILE_NTLS, SSL_R_X509_LIB);
+        goto end;
+    }
+
+    if (EVP_PKEY_id(pkey) == EVP_PKEY_SM2
+        || EVP_PKEY_id(pkey) == EVP_PKEY_EC)
```
Comment (InfoHunter) [inline ssl/ssl_rsa.c:173]: 为啥还要判断或者是PKEY_EC？

File: ssl/s3_lib.c (hunk 1 above)
Comment (dongbeiouba) [reply ssl/s3_lib.c:2960]: GB对应国标，套件定义来源于TLCP标准

File: ssl/ssl_rsa.c (hunk 2 above)
Comment (dongbeiouba) [reply ssl/ssl_rsa.c]: 已经修改，去掉了EVP_PKEY_EC

**Review by InfoHunter — APPROVED** [review]
Re-approve

File: include/openssl/ntls.h [hunk 3]
```diff
@@ -5,58 +5,82 @@
...
+#  endif
+
 /* NTLS version */
-# define NTLS1_1_VERSION	0x0101
-# define NTLS1_1_VERSION_MAJOR	0x01
-# define NTLS1_1_VERSION_MINOR	0x01
-# define NTLS_VERSION		NTLS1_1_VERSION
-# define NTLS_VERSION_MAJOR	NTLS1_1_VERSION_MAJOR
-# define NTLS_VERSOIN_MINOR	NTLS1_1_VERSION_MINOR
+#  define NTLS1_1_VERSION	    0x0101
```
Comment (jinjiu) [inline include/openssl/ntls.h:16]: 16行和19行没有上下对齐

Comment (dongbeiouba) [reply include/openssl/ntls.h]: 已修改

**Pull request merged**